File: Week 1 - C/cash/cash.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int change;

    // Solicitar valor de troco ao usuário até que a entrada seja válida
    do
    {
        change = get_int("Change owed: ");
    }
    while (change < 1);

    // Inicializar contador de moedas
    int coins = 0;

    // Contar o número mínimo de moedas necessárias
    while (change >= 25)
    {
        change -= 25;
        coins++;
    }

    while (change >= 10)
    {
        change -= 10;
        coins++;
    }

    while (change >= 5)
    {
        change -= 5;
        coins++;
    }

    while (change >= 1)
    {
        change -= 1;
        coins++;
    }

    // Imprimir o número mínimo de moedas necessárias
    printf("%d\n", coins);
}

File: week6/cash.py
from cs50 import get_float, get_int

while True:
    change = get_float("Change owed: ")
    if change >= 0.01:
        break

change = round(change * 100) #converte para centavos

coins = 0

while change >= 25:
    change -= 25
    coins += 1

while change >= 10:
    change -= 10
    coins += 1

while change >= 5:
    change -= 5
    coins += 1

while change >= 1:
    change -= 1
    coins += 1

print(f"{coins}")

